service/internal/integrationtest/s3/checksum_test.go

```diff
@@ -19,9 +19,38 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
+var (
+	isInitialCall = true
```

Comment:
Make this a field on `retryClient`, configuring a test to retry will only work the one time if it's package-level.